superset/db_engine_specs.py

```diff
@@ -804,6 +809,21 @@ class PrestoEngineSpec(BaseEngineSpec):
             date_add('day', 1, CAST({col} AS TIMESTAMP))))",
     }
 
+    type_map = {
+        'boolean': types.Boolean,
+        'tinyint': mysql.MSTinyInteger,
```

Comment:
`mysql` ... just calling it out as a divergence from the impl agnostic types that all of the others are using. It feels "leaky".

Reply:
Created skeleton types and moved types to its own file.

Reply:
Looks like skeleton types are now causing CI failures:

```
AttributeError: 'GenericTypeCompiler' object has no attribute 'visit_ARRAY'
During handling of the above exception, another exception occurred:
... lots of stuff ...
sqlalchemy.exc.UnsupportedCompilationError: Compiler <sqlalchemy.sql.compiler.GenericTypeCompiler object at 0x7f9f6189ed68> can't render element of type <class 'superset.models.sql_types.presto_sql_types.ARRAY'>
```

As annoying as it would be to revert the types you added (and going back to the original question I sort of raised) I wonder if it is a viable path to just leave it with the mysql type there? Obviously it would be less "correct" than having a well-defined presto dialect but...

@john-bodley @graceguo-supercat @michellethomas @mistercrunch Does that value get passed around to other bits of code that might actually care if it is _actually_ a mysql column? How bad is it to use the only existing built-in representation of tinyint here, under the mysql dialect?

Reply:
Talked to Dave offline. I fixed the CI issues. So we can continue to have our skeleton types and not have to rely on other db classes like mysql.